File: ZezexJava/server/src/Game/Skill/Skills.java
package Game.Skill;

import Game.*;

public class Skills {
	private long RoleId;
	public final long getRoleId() {
		return RoleId;
	}
	private BSkills Bean;

	public Skills(long roleId, BSkills bean) {
		this.RoleId = roleId;
		this.Bean = bean;
	}

	public final Skill GetSkill(int id) {
		var skill = Bean.getSkills().get(id);
		if (null != skill) {
			var extraTypeId = skill.getExtra().getTypeId();
			if (extraTypeId == BSkillAttackExtra.TYPEID)
				return new SkillAttack(skill, (BSkillAttackExtra)skill.getExtra().getBean());
			throw new RuntimeException("unknown extra");
		}
		return null;
	}
}

File: Sample/server/Game/Skill/Skills.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Game.Skill
{
    public class Skills
    {
        public long RoleId { get; }
        private BSkills Bean;

        public Skills(long roleId, BSkills bean)
        {
            this.RoleId = roleId;
            this.Bean = bean;
        }

        public Skill GetSkill(int id)
        {
            if (Bean.Skills.TryGetValue(id, out var skill))
            {
                switch (skill.Extra.TypeId)
                {
                    case BSkillAttackExtra.TYPEID: return new SkillAttack(skill, (BSkillAttackExtra)skill.Extra.Bean);
                    default:
                        throw new System.Exception("unknown extra");
                }
            }
            return null;
        }
    }
}
